client-gen/src/lib.rs

```diff
@@ -4,7 +4,10 @@ use anyhow::{Context, Result};
 use convert_case::{Case, Casing};
 use std::{ffi::OsStr, fs, path::Path};
 
-pub fn generate_client_from_idl(idl_path: &Path, out_path: &Path) -> Result<()> {
+pub fn generate_client_from_idl<P: AsRef<Path>>(idl_path: P, out_path: P) -> Result<()> {
```

Comment:
This code requires P to be the same type for both arguments which might not be the case, so it is better to write
```suggestion
pub fn generate_client_from_idl(idl_path: impl AsRef<Path>, out_path: impl AsRef<Path>) -> Result<()> {
```